src/GraphQL/Reflection/ReflectionHelper.cs

```diff
@@ -98,11 +98,13 @@ public static PropertyInfo PropertyForField(this Type type, string field)
             }
 
             if (parameters.Length > index
-                && context.UserContext != null
-                && context.UserContext?.GetType() == parameters[index].ParameterType)
+                && context.UserContext != null)
             {
-                arguments[index] = context.UserContext;
-                index++;
+                if (parameters[index].ParameterType.IsInstanceOfType(context.UserContext))
```

Comment:
Sometimes I wonder at the names of the methods. It would have to be called `IsTypeOfInstance`.